settings.php

```diff
@@ -249,4 +251,54 @@
 
     $settings->add(new admin_setting_configcheckbox('tool_objectfs/preferexternal',
         new lang_string('settings:preferexternal', 'tool_objectfs'), '', ''));
+
+    // Tagging settings.
+    $settings->add(new admin_setting_heading('tool_objectfs/taggingsettings',
+        new lang_string('settings:taggingheader', 'tool_objectfs'),
+        get_string('settings:tagging:help', 'tool_objectfs')
+    ));
+
+    $settings->add(new admin_setting_configcheckbox('tool_objectfs/taggingenabled',
+        new lang_string('settings:taggingenabled', 'tool_objectfs'), '', 0));
+
+    $settings->add(new admin_setting_description('tool_objectfs/tagsources',
+        new lang_string('settings:tagsources', 'tool_objectfs'),
+        tag_manager::get_tag_source_summary_html()
+    ));
+
+    $settings->add(new admin_setting_configtext('tool_objectfs/maxtaggingperrun',
+        new lang_string('settings:maxtaggingperrun', 'tool_objectfs'),
+        get_string('settings:maxtaggingperrun:desc', 'tool_objectfs'),
+        10000,
+        PARAM_INT
+    ));
+
+    $settings->add(new admin_setting_configtext('tool_objectfs/maxtaggingiterations',
+        new lang_string('settings:maxtaggingiterations', 'tool_objectfs'),
+        get_string('settings:maxtaggingiterations:desc', 'tool_objectfs'),
+        1000,
+        PARAM_INT
+    ));
+
+    $settings->add(new admin_setting_configcheckbox('tool_objectfs/overwriteobjecttags',
+        new lang_string('settings:overrideobjecttags', 'tool_objectfs'),
+        get_string('settings:overrideobjecttags:desc', 'tool_objectfs'),
+        1
+    ));
+
+    // Tagging status.
+    $settings->add(new admin_setting_heading('tool_objectfs/taggingstatus',
+        new lang_string('settings:taggingstatus', 'tool_objectfs'), ''));
+
+    // Build overview of tag statuses.
+    $settings->add(new admin_setting_description('tool_objectfs/taggingstatuscounts',
+        new lang_string('settings:taggingstatuscounts', 'tool_objectfs'),
+        tag_manager::get_tag_sync_status_summary_html()
+    ));
```

Comment:
Fair call, I decided to move it to a check api class, and then just link them from the admin settings, which has the added benefit of if they warn about failures, they will alert the admins (e.g. via heartbeat)

It would be super nice if https://tracker.moodle.org/browse/MDL-67898 was available in lower versions so we could put the check straight into the settings but since its 4.4+ and that is somewhat far away from the supported 3.10 branch I decided to against it (e.g. gated behind `class_exists`)